lib/Cake/View/Helper/FormHelper.php

```diff
@@ -1638,7 +1645,12 @@ public function radio($fieldName, $options = array(), $attributes = array()) {
 			$between = '';
 		}
 		if ($legend) {
-			$out = $this->Html->useTag('fieldset', '', $this->Html->useTag('legend', $legend) . $between . $out);
+			if (is_string($fieldset)) {
+				$fieldsetClass = sprintf(' class="%s"', $fieldset);
```

Comment:
> Should I use $fieldsetClass = sprintf(' class="%s"', h($fieldset)); instead?

That, or `$fieldsetAttrs = ['class' => $fieldset];` and pass `$fieldsetAttrs` to `useTag()` when creating the fieldset element.